### PR TITLE
Fix Kohya UNet LoRA key conversion for conv_in/conv_out/time_embedding

### LINKED ISSUE #14005: _convert_unet_lora_key misses conv_in/conv_out/time_embedding mapping for Kohya UNet LoRAs
Note: I've been running some automated LoRA loading tests here https://github.com/Nerogar/OneTrainer/issues/1544
Many issues on OneTrainer's side to fix, but where I think it's a diffusers issue I'm reporting it upstream, but these can be about old or even obsolete features such as support Kohya-format LoRAs for SD 1.5 as a result.

Feel free to close if you think an issue isn't worth reporting anymore.

### Describe the bug

Loading a Kohya-format UNet LoRA that touches `conv_in`, `conv_out`, or `time_embedding.linear_1`/`linear_2` produces unexpected keys and those layers' weights aren't applied. Reproduced on SD 1.5, SD 2.1, and SDXL.

`_convert_unet_lora_key()` (`diffusers/loaders/lora_conversion_utils.py`) converts Kohya-style keys by replacing every `_` with `.` and then patching specific substrings back (`input.blocks`→`down_blocks`, `proj.in`→`proj_in`, `conv.shortcut`→`conv_shortcut`, etc.):

https://github.com/huggingface/diffusers/blob/0f1abc4ae8b0eb2a3b40e82a310507281144c423/src/diffusers/loaders/lora_conversion_utils.py#L260-L313

There's no patch for:
- `conv.in` → `conv_in`
- `conv.out` → `conv_out`
- `time.embed.0` / `time.embed.2` → `time_embedding.linear_1` / `time_embedding.linear_2`

So a Kohya key like `lora_unet_conv_in.lora_A.weight` comes out as `conv.in.lora_A.weight` instead of `conv_in.lora_A.weight`, and the PEFT adapter loader reports it as an unexpected key instead of applying it to the model's `conv_in` module.

### Reproduction

```python
import torch
from diffusers import DiffusionPipeline

pipe = DiffusionPipeline.from_pretrained("stable-diffusion-v1-5/stable-diffusion-v1-5", torch_dtype=torch.bfloat16)
pipe.load_lora_weights("path/to/kohya_format_lora.safetensors")
```

```
Loading adapter weights from state_dict led to unexpected keys found in the model: conv.in.lora_A.default_0.weight, conv.in.lora_B.default_0.weight, conv.out.lora_A.default_0.weight, conv.out.lora_B.default_0.weight, time.embedding.linear.lora_A.default_0.weight, time.embedding.linear.lora_B.default_0.weight.
```

Same result on `sd2-community/stable-diffusion-2-1` and `stabilityai/stable-diffusion-xl-base-1.0`.

### Logs

```
Loading adapter weights from state_dict led to unexpected keys found in the model: conv.in.lora_A.default_0.weight, conv.in.lora_B.default_0.weight, conv.out.lora_A.default_0.weight, conv.out.lora_B.default_0.weight, time.embedding.linear.lora_A.default_0.weight, time.embedding.linear.lora_B.default_0.weight.
```

### System Info

- diffusers commit: 0f1abc4ae8b0eb2a3b40e82a310507281144c423 (main)
- Python: 3.12
- Platform: Linux

### Who can help?

@yiyixuxu @sayakpaul

### PR DESCRIPTION
**Pull request by dxqb**
Fixes #14005.

It is in principle related to https://github.com/huggingface/diffusers/pull/14080, as it addresses layers that aren't trained that often, but *can* be trained.

### What does this PR do?

Kohya-format UNet LoRA keys for several top-level UNet submodules weren't being
  converted to diffusers names, so they didn't match any parameter and were reported as
  unexpected keys instead of being applied. This covers both UNet dialects kohya-ss
  trains on: the diffusers UNet (SD 1.x) and the sgm/LDM UNet (SDXL).

  `_convert_unet_lora_key` — add the missing name patches:
  - `conv.in → conv_in`, `conv.out → conv_out`
  - sgm `input.blocks.0.0 → conv_in`, `out.2 → conv_out` (mapped before the block renames
    so `input_blocks.0.0` isn't mistaken for a down-block)
  - `time.embed.0/.2 → time_embedding.linear_1/2` (sgm) and
    `time.embedding.linear.1/2 → time_embedding.linear_1/2` (diffusers)
  - `label.emb.0.0/0.2 → add_embedding.linear_1/2` (SDXL added-conditioning MLP)

  `_maybe_map_sgm_blocks_to_diffusers` — pass top-level sgm modules
  (`time_embed`, `label_emb`, `out`, `input_blocks.0.0`) through unchanged so the key
  converter handles them, instead of block-remapping or hitting the "layer not supported"
  raise.

### Who can review?

PEFT: @sayakpaul @BenjaminBossan

### CODE REVIEW

**Comment by dxqb** [comment]
needs a test, I'll mark the PR as ready as soon as that passes

**Comment by dxqb** [comment]
this PR is ready for review now

**Review by sayakpaul — APPROVED** [review]
Thanks for the massive contribution!
